### PR TITLE
[IMP] {hr,sale}_timesheet, _*: make separate method for access

### PR DESCRIPTION
**Pull request by kcv-odoo**
_* = project_timesheet_holidays

Before this PR checking validation for creating and writing on timesheet done in create and write method which makes customization difficult for user.

So, in this PR check validation for creating and writing on timesheet done in separate method _check_can_create and _check_can_write respectively for timesheet.

task-2928942

### CODE REVIEW

**Review by auon-odoo — APPROVED** [review]
Lgtm! Thanks for this neat work :dart:

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left 2 comments and after that it will look good to me. 🙂

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂 

robodoo r+

File: addons/project_timesheet_holidays/models/account_analytic.py [hunk 1]
```diff
@@ -36,16 +36,12 @@ def _unlink_except_linked_leave(self):
...
-            task_ids = [vals['task_id'] for vals in vals_list if vals.get('task_id')]
-            has_timeoff_task = self.env['project.task'].search_count([('id', 'in', task_ids), ('is_timeoff_task', '=', True)], limit=1) > 0
-            if has_timeoff_task:
-                raise UserError(_('You cannot create timesheets for a task that is linked to a time off type. Please use the Time Off application to request new time off instead.'))
-        return super().create(vals_list)
-
-    def write(self, vals):
-        if not self.env.su and self.filtered('holiday_id'):
+    def _check_can_write(self, values):
+        if not self.env.su and any(line.holiday_id for line in self):
```
[the remark this answers is not in the document]
Comment (kcv-odoo) [reply addons/project_timesheet_holidays/models/account_analytic.py]: Done! Thanks :)

**Pull request closed without merging**